src/modules/authentication/AuthenticationWrapper.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  AccountInfo,
+  AuthenticationResult, InteractionRequiredAuthError,
+  PopupRequest, SilentRequest
+} from '@azure/msal-browser';
+
+import { AUTH_URL, DEFAULT_USER_SCOPES } from '../../app/services/graph-constants';
+import { geLocale } from '../../appLocale';
+import { getCurrentUri } from './authUtils';
+import IAuthenticationWrapper from './IAuthenticationWrapper';
+import { msalApplication } from './msal-app';
+
+const defaultScopes = DEFAULT_USER_SCOPES.split(' ');
+const homeAccountKey = 'home';
+
+export class AuthenticationWrapper implements IAuthenticationWrapper {
+
+  private static instance: AuthenticationWrapper;
+
+  public static getInstance(): AuthenticationWrapper {
+    if (!AuthenticationWrapper.instance) {
+      AuthenticationWrapper.instance = new AuthenticationWrapper()
+    }
+    return AuthenticationWrapper.instance
+  }
+
+  public getSessionId() {
+    const account = this.getAccount();
+    const idTokenClaims: any = account?.idTokenClaims;
+    return idTokenClaims?.sid;
+  }
+
+  public async logIn(sessionId = ''): Promise<AuthenticationResult> {
+    try {
+      return await this.getAuthResult([], sessionId);
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  public logOut() {
+    msalApplication.logout();
+  }
+
+  public async logOutPopUp() {
+    const endSessionEndpoint = (await msalApplication.getDiscoveredAuthority()).endSessionEndpoint;
+    (window as any).open(endSessionEndpoint, 'msal', 400, 600);
+  }
+
+  /**
+ * Generates a new access token from passed in scopes
+ * @param {string[]} scopes passed to generate token
+ *  @returns {Promise.<AuthenticationResult>}
+ */
+  public async acquireNewAccessToken(scopes: string[] = []): Promise<AuthenticationResult> {
+    try {
+      const authResult = await this.getAuthResult(scopes);
+      return authResult;
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  private getAccount(): AccountInfo | undefined {
+    if (msalApplication) {
+      const allAccounts = msalApplication.getAllAccounts();
+      if (allAccounts && allAccounts.length > 0) {
+        if (allAccounts.length > 1) {
+          const homeAccountId = this.getHomeAccountId();
+          const account = (homeAccountId) ? msalApplication.getAccountByHomeId(homeAccountId) : allAccounts[0];
```

Comment:
If the home account id is undefined,  I feel it is better to throw an error or ask the user to relogin. Returning `allAccounts[0]` feels more like an assumption that we are making.